### PR TITLE
Three new fields for Snapshot restore jobs options

### PR DESCRIPTION
**Pull request by coderGo93**
Added new three fields for Snapshot restore jobs called `oplogTs`, `oplogInc` and `pointInTimeUTCSeconds` and its test

### CODE REVIEW

**Comment by themantissa** [comment]
@gssbzn added you for a review as an FYI and check in case it's something you can use for mCLI.

File: mongodbatlas/cloud_provider_snapshot_restore_jobs.go [hunk 1]
```diff
@@ -30,19 +30,21 @@ var _ CloudProviderSnapshotRestoreJobsService = &CloudProviderSnapshotRestoreJob
...
+	CreatedAt             string   `json:"createdAt,omitempty"`             // UTC ISO 8601 formatted point in time when Atlas created the restore job.
+	Expired               bool     `json:"expired,omitempty"`               // Indicates whether the restore job expired.
+	ExpiresAt             string   `json:"expiresAt,omitempty"`             // UTC ISO 8601 formatted point in time when the restore job expires.
+	FinishedAt            string   `json:"finishedAt,omitempty"`            // UTC ISO 8601 formatted point in time when the restore job completed.
+	Links                 []*Link  `json:"links,omitempty"`                 // One or more links to sub-resources and/or related resources. The relations between URLs are explained in the Web Linking Specification.
+	Timestamp             string   `json:"timestamp,omitempty"`             // Timestamp in ISO 8601 date and time format in UTC when the snapshot associated to snapshotId was taken.
+	OplogTs               string   `json:"oplogTs,omitempty"`               // If your cluster has Point-in-Time restores enabled, you can specify a timestamp from the oplog from which to restore a backup.
```
Comment (gssbzn) [inline mongodbatlas/cloud_provider_snapshot_restore_jobs.go:46]: had to go and compare our docs to our code, they both should be int64, and for your test the sample should be something like `1536610288`

Comment (gssbzn) [reply mongodbatlas/cloud_provider_snapshot_restore_jobs.go:46]: We may need to file a ticket with the docs team to fix this cc @themantissa

Comment (gssbzn) [inline mongodbatlas/cloud_provider_snapshot_restore_jobs.go:46]: Another thing here, if you provide `oplogTs` you **must** also provide an `oplogInc` (int)
this also looks to be missing from the docs

Comment (themantissa) [reply mongodbatlas/cloud_provider_snapshot_restore_jobs.go:46]: @gssbzn huge thank you for catching this!  I'll get a docs ticket in.

Comment (coderGo93) [reply mongodbatlas/cloud_provider_snapshot_restore_jobs.go:46]: I understand, gustavo, I'll change it and thank you very much :+1:

Comment (coderGo93) [reply mongodbatlas/cloud_provider_snapshot_restore_jobs.go:46]: I see, makes sense, gustavo, thank you very much, I'll change it

File: mongodbatlas/cloud_provider_snapshot_restore_jobs.go [hunk 2]
```diff
@@ -30,19 +30,21 @@ var _ CloudProviderSnapshotRestoreJobsService = &CloudProviderSnapshotRestoreJob
...
-	Expired           bool     `json:"expired,omitempty"`           // Indicates whether the restore job expired.
-	ExpiresAt         string   `json:"expiresAt,omitempty"`         // UTC ISO 8601 formatted point in time when the restore job expires.
-	FinishedAt        string   `json:"finishedAt,omitempty"`        // UTC ISO 8601 formatted point in time when the restore job completed.
-	Links             []*Link  `json:"links,omitempty"`             // One or more links to sub-resources and/or related resources. The relations between URLs are explained in the Web Linking Specification.
-	Timestamp         string   `json:"timestamp,omitempty"`         // Timestamp in ISO 8601 date and time format in UTC when the snapshot associated to snapshotId was taken.
+	ID                    string   `json:"id,omitempty"`                    // The unique identifier of the restore job.
+	SnapshotID            string   `json:"snapshotId,omitempty"`            // Unique identifier of the snapshot to restore.
+	DeliveryType          string   `json:"deliveryType,omitempty"`          // Type of restore job to create. Possible values are: automated or download
```
Comment (themantissa) [inline mongodbatlas/cloud_provider_snapshot_restore_jobs.go:35]: May also want to update the comment - there are 3 types now, this is the newest: pointInTime

File: mongodbatlas/cloud_provider_snapshot_restore_jobs_test.go [hunk 3]
```diff
@@ -161,8 +163,10 @@ func TestCloudProviderSnapshotRestoreJobs_List(t *testing.T) {
 						Rel:  "http://mms.mongodb.com/group",
 					},
 				},
-				SnapshotID: "5b6211ff87d9d663c59d3feg",
-				Timestamp:  "2018-08-01T20:02:07Z",
+				SnapshotID:            "5b6211ff87d9d663c59d3feg",
+				Timestamp:             "2018-08-01T20:02:07Z",
+				OplogTs:               "2018-08-01T20:02:07Z",
+				PointInTimeUTCSeconds: 1583753751,
```
Comment (themantissa) [inline mongodbatlas/cloud_provider_snapshot_restore_jobs_test.go:169]: Also fyi for later be sure when supporting in Terraform you note that deliveryType for doing a point in time restore job (PIT) would be pointInTime also note in the docs that you must specify either pointInTimeUTCSeconds or oplogTs, but not both.

File: mongodbatlas/cloud_provider_snapshot_restore_jobs.go (hunk 2 above)
Comment (coderGo93) [reply mongodbatlas/cloud_provider_snapshot_restore_jobs.go:35]: You're right, I'll change it, thank you melissa

**Pull request merged**